### ISSUE TITLE
Headless BeEF

### ISSUE THREAD

**Issue by bcoles** [beefproject/beef#1389]
* [x] Replace existing UI calls to controllers with calls to REST API:
  * [x] Proxy
  * [x] Logs
  * [x] XssRays
  * [x] Requester
* [x] Ensure web admin UI interface can be safely disabled; allowing BeEF to run headless, managed only via the RESTful API.
* [x] Move the `web_ui_basepath` option out of the main `config.yaml` and into the Admin UI `config.yaml`.

Optionally, also allow the REST API to run on a different port. Client-side assets such as the hook currently share the same port as the web admin UI and the REST interface.

~Blocked on @antisnatchor 's BeEF-NG branch to make BeEF web server agnostic.~ Unfortunately, this isn't going to be implemented.

**Comment by bcoles** [comment]
This is mostly done for the core.

There's a bunch of functionality relating to the command modules which should probably be reworked. We can probably live without it, as much of this functionality is already available via the REST API.

The Requester code also needs to be reviewed. We can probably live without this too, as the proxy extension offers similar functionality.

The only remaining Admin UI controllers are `panel`, `authentication` and `modules`.

The `panel` and `authentication` controllers are required for the admin UI and aren't required anywhere else.

BeEF works with `admin_ui.enable: false`.

Unfortunately moving the REST API to a separate port is not going to happen any time soon. Fortunately, access to the API can be restricted by IP address. Unfortunately, the access controls can be bypassed (#1354).

Next up is to create a standalone REST client (#1390) for remote management. The existing REST interface is a bit of a mess and contains lots of duplicated code. We'll burn that bridge when we get to it.

It would be nice to reinstate the local console extension at some point too.

It would also be nice to have a single page application, similar to the existing admin ui, offering a web interface which can connect to a user-specified running BeEF instance. This is kind of tedious, firstly because all the assets are located on the BeEF server, and secondly because I don't to write a SPA.

**Comment by DeezyE** [comment]
This is closed? I thought headless Beef was a logical future direction? Is this captured better elsewhere?

**Comment by bcoles** [comment]
> This is closed?

This issue is closed.

> I thought headless Beef was a logical future direction?

The admin UI is no longer a requirement. BeEF can be operated using only the REST API.

>  Is this captured better elsewhere?

There are still many things to be done:

https://github.com/beefproject/beef/issues/1389#issuecomment-464354068
